### PR TITLE
D8CORE-5955: Added journal publisher field, updated publisher label.

### PR DESCRIPTION
**Pull request by imonroe**
# READY FOR REVIEW

# Summary
- Journals need a Journal Publisher field.  Up until now, users have been using the "Publisher" field as the journal name, so we update the label on that field to make it clear how it's used, and we add a Journal Publisher field to hold the publisher of the journal (e.g., Elsevier, IEEE, etc.)

# Review By (Date)
- 3.1.0 code cutoff.

# Criticality
- low

# Urgency
- normal

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Rebuild Cache and import config `drush cr ; drush ci`
3. Create a publication with a Journal citation.
4. Verify that you can store (and view) the journal title and the journal publisher on the Publication page.  The Citation styles don't change in list views.

### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory? YES

## Front End Validation
- [ ] Design is approved by @ user?
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: config/sync/field.field.citation.su_article_journal.su_journal_publisher.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+langcode: en
+status: true
+dependencies:
+  config:
+    - field.storage.citation.su_journal_publisher
+    - stanford_publication.citation_type.su_article_journal
+id: citation.su_article_journal.su_journal_publisher
+field_name: su_journal_publisher
+entity_type: citation
+bundle: su_article_journal
```
Comment (pookmish) [inline config/sync/field.field.citation.su_article_journal.su_journal_publisher.yml:11]: All citation fields aren't prefixed with a bundle. it makes it easier to build the citation for the different bundles.
Also I don't see this field getting added to the citation format. Is that intentional?

**Review by pookmish — CHANGES_REQUESTED** [review]

File: config/sync/field.field.citation.su_article_journal.su_journal_publisher.yml (hunk 1 above)
Comment (imonroe) [reply config/sync/field.field.citation.su_article_journal.su_journal_publisher.yml:11]: That's correct, the ticket did not request any change to the citation formats, so I left well enough alone.

Comment (imonroe) [reply config/sync/field.field.citation.su_article_journal.su_journal_publisher.yml:11]: I didn't want to change the machine name for `su_publisher`, to preserve backwards compatibility, so I made it `su_journal_publisher` to differentiate the new field.  It's only used in the Journal citation type.

**Review by pookmish — APPROVED** [review]
Meh, I'm good with it as is.

**Pull request merged**